app/controllers/application_controller.rb

```diff
@@ -209,6 +209,11 @@ def render_hidden(template='request/hidden', opts = {})
   # URL using the first three digits of the info request id, because we can't
   # have more than 32,000 entries in one directory on an ext3 filesystem.
   def foi_fragment_cache_part_path(param)
+    if rails5?
+      param =
+        param.permit(:incoming_message_id, :part, :file_name, :id,
+                     :only_path, :locale, :skip_cache)
```

Comment:
Unclear where these params are sourced from. If I understand correctly they are from https://github.com/mysociety/alaveteli/blob/develop/config/routes.rb#L123-L130 https://github.com/mysociety/alaveteli/blob/develop/app/controllers/request_controller.rb#L609 and https://github.com/mysociety/alaveteli/blob/develop/app/controllers/request_controller.rb#L679 maybe the commit message could be expanded.

Reply:
I think it was finalised by setting a breakpoint and watching what was passed in as a double-check but I've expanded the commit message as follows:

```
The permitted parameter list is derived from:
  We generally expect :locale so that's included here
  The get_attachment and get_attachment_as_html routes specify
    :id, :incoming_message_id, :part, and :file_name
  RequestContoller#cache_attachments
    expects :skip_cache, :only_path
```